contracts/src/CrossChainApplications/NativeTokenBridge/INativeTokenDestination.sol

```diff
@@ -0,0 +1,45 @@
+// (c) 2023, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+// SPDX-License-Identifier: Ecosystem
+
+pragma solidity 0.8.18;
+
+/**
+ * @dev Interface that describes functionalities for a contract that can mint native tokens when
+ * paired with a NativeTokenSource contract that will lock tokens on another chain.
+ */
+interface INativeTokenDestination {
+    /**
+     * @dev Emitted when tokens are burned in this bridge contract to be bridged to the source chain.
+     */
+    event TransferToSource(
+        address indexed sender,
+        address indexed recipient,
+        uint256 amount,
+        address feeContractAddress,
+        uint256 feeAmount,
+        uint256 teleporterMessageID
+    );
+
+    /**
+     * @dev Emitted when tokens are not minted in order to collateralize the source contract.
+     */
+    event CollateralAdded(uint256 amount, uint256 remaining);
+
+    /**
+     * @dev Emitted when minting native tokens.
```

Comment:
Just to confirm, the native asset of a chain for example BTC, ETH are referred to as tokens, but also erc20 tokens? Wasn't sure if there's a more clear way to differentiate the native gas token from erc20 tokens, or if that's enough already.